### PR TITLE
Document zmq-messageQ

### LINKED ISSUE #1109: zmq-messageQ code documentation
### Description 
Write code comments for the following:
zmq/MessageQ.java

### PR DESCRIPTION
**Pull request by karimodm**
Documents zmq/MessageQ

Fixes #1109

### CODE REVIEW

**Comment by karimodm** [comment]
I had verified it was up to date, but I have now amended it with some extra information, and removed the WIP comment.

File: src/main/java/com/iota/iri/zmq/README.md [hunk 1]
```diff
@@ -22,19 +20,22 @@ A client interested in tip selection metrics may subscribe to `mctn`, short for
...
 * `hmr` for the hit to miss ratio
 * `antn` for added non-tethered neighbors ( testnet only )
 * `rntn` for refused non-tethered neighbors
+* `rstat` for information about the tips requester
 * `rtl` for transactions randomly removed from the request list
 * `lmi` for the latest milestone index
 * `lmsi` for the latest solid milestone index
 * `lmhs` for the latest solid milestone hash
 * `sn` for newly confirmed transactions ( by solid milestone children measurement )
 * `tx` for newly seen transactions
```
Comment (alon-e) [inline src/main/java/com/iota/iri/zmq/README.md]: missing `tx_trytes`: [ZmqPublishProvider](https://github.com/iotaledger/iri/blob/dev/src/main/java/com/iota/iri/storage/ZmqPublishProvider.java#L97-L99)

File: src/main/java/com/iota/iri/zmq/MessageQ.java [hunk 2]
```diff
@@ -11,7 +11,25 @@
...
 /**
- * Created by paul on 6/20/17.
+ * Allows publishing of IRI events to a ZeroMQ queue.
+ *
+ * <p>
+ *    In order to be able to watch IRI events, and possibly build tools on top of them, this class
+ *    allows the code to publish operational facts to a ZMQ stream.
+ * </p>
+ * <p>
+ *    IRI events are published to this queue prepending the topic of the event to the body
```
Comment (alon-e) [inline src/main/java/com/iota/iri/zmq/MessageQ.java]:
```suggestion
 *    IRI events are published to this queue prepending the {@code topic} of the event to the body
```

File: src/main/java/com/iota/iri/zmq/MessageQ.java [hunk 3]
```diff
@@ -11,7 +11,25 @@
...
+ *    of the message. The topic describes the type or the source of the event and is represented by
```
Comment (alon-e) [inline src/main/java/com/iota/iri/zmq/MessageQ.java]:
```suggestion
 *    of the {@code message}. The topic describes the type or the source of the event and is represented by
```

File: src/main/java/com/iota/iri/zmq/MessageQ.java [hunk 4]
```diff
@@ -11,7 +11,25 @@
...
+ *    a short lowercase string.
+ *    Some example topics:
+ *    <ol>
+ *        <li>tx: transactions events</li>
```
Comment (alon-e) [inline src/main/java/com/iota/iri/zmq/MessageQ.java]: format the topics as code?

File: src/main/java/com/iota/iri/zmq/MessageQ.java [hunk 5]
```diff
@@ -11,7 +11,25 @@
...
+ *        <li>lm: milestones events</li>
+ *        <li>dns: neighbors' DNS events</li>
+ *    </ol>
+ *    To monitor for activity on a specific address, the topic is instead the Address to watch.
```
Comment (alon-e) [inline src/main/java/com/iota/iri/zmq/MessageQ.java]:
```suggestion
 *    To monitor for activity on a specific address, the topic is instead the {@code Address} to watch.
```

File: src/main/java/com/iota/iri/zmq/MessageQ.java [hunk 6]
```diff
@@ -26,6 +44,14 @@ public static MessageQ createWith(ZMQConfig config) {
         return new MessageQ(config.getZmqPort(), config.getZmqIpc(), config.getZmqThreads(), config.isZmqEnabled());
     }
 
+    /**
+     * Creates and starts a ZMQ publisher.
+     *
+     * @param port port the publisher will be bound to
+     * @param ipc IPC socket to bind the publisher to
```
Comment (alon-e) [inline src/main/java/com/iota/iri/zmq/MessageQ.java]:
```suggestion
     * @param ipc IPC socket the publisher will be bound to
```
or swap the `port` doc to this

File: src/main/java/com/iota/iri/zmq/MessageQ.java [hunk 7]
```diff
@@ -26,6 +44,14 @@ public static MessageQ createWith(ZMQConfig config) {
...
+     * @param nthreads number of threads used by the ZMQ publisher
+     * @param enabled boolean enable flag; defaults to "false"
```
Comment (alon-e) [inline src/main/java/com/iota/iri/zmq/MessageQ.java]: if set to default, ZMQ will not run - won't bind etc.

File: src/main/java/com/iota/iri/zmq/MessageQ.java [hunk 8]
```diff
@@ -41,13 +67,22 @@ private MessageQ(int port, String ipc, int nthreads, boolean enabled) {
         }
     }
 
+    /**
+     * Publishes an event to the queue
+     *
+     * @param message message body, prepended by topic string
+     * @param objects objects to be rendered as part of the message
```
Comment (alon-e) [inline src/main/java/com/iota/iri/zmq/MessageQ.java]: not sure "to be rendered" is the best term, but I don't have a better one now.
from String.format: 
```
args - Arguments referenced by the format specifiers in the format string
```

**Pull request merged**